### PR TITLE
Refactor package structure - naming

### PR DESCRIPTION
**Pull request by dhruvaa02**
## Context

Wanted to refactor our code to comply better with clean architecture and to be easier to follow along.

## Your Changes

1. Seperated DataAccess and DataProcess
2. Renamed the packages to the meeting outline.
3. Removed redundant default constructors in Fetcher and Recommender.

**Description**:

**Type of change** (select all that apply):

- [x] Refactoring (changing code structure only)

## Testing

All automated unit tests passed and behave as it used to.

## Checklist

- [x] Code is up to date with main branch (no merge conflicts)
- [x] Code has been properly tested
- [x] Code has been properly documented
- [ ] One or two people have reviewed this pull request

### CODE REVIEW

**Review by nguya03584 — APPROVED** [review]

File: src/test/java/APIBoundaryTest/ApplicationTest.java [hunk 1]
```diff
@@ -1,7 +1,6 @@
 package APIBoundaryTest;
 
-import com.aviva.APIBoundary.Application;
-import org.junit.jupiter.api.BeforeEach;
```
Comment (leenlab2) [inline src/test/java/APIBoundaryTest/ApplicationTest.java:4]: why did you remove the Beforeeach?

**Review by leenlab2 — COMMENTED** [review]
left one comment but other than that its good

File: src/test/java/APIBoundaryTest/ApplicationTest.java (hunk 1 above)
Comment (dhruvaa02) [reply src/test/java/APIBoundaryTest/ApplicationTest.java:4]: It was actually gone, even before this PR so not sure why it's coming up in this PR. In that file, I don't do any repeated setup or anything because there is just one test so it was redundant.
